dashboard-api/internal/platform/storage/cockroachdb/study_segments_repository_test.go

```diff
@@ -0,0 +1,146 @@
+package cockroachdb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+)
+
+func Test_StudySegmentsRepository_GetAllTimeSegmentsProgress_UnexpectedError_during_Query(t *testing.T) {
+	studyId := "c463c577-a3d1-482e-b638-10e6733e2325"
+	db, sqlMock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+	require.NoError(t, err)
+
+	sqlMock.ExpectQuery(
+		"SELECT created, details FROM optimization_reports WHERE report_type = 'FACEBOOK_OPTIMIZATION' and study_id = $1 ORDER BY created ASC").
```

Comment:
I'm not sure I see the advantage of this sql mock library. Either I want to have integration tests to test everything with the database included, or I want unit tests where I mock that we can handle edge cases. Why do I care what the query is if I anyways want to test the query does what I want? It's brittle, you have to change the tests when the query changes, even if the rest of your business logic and desired behavior is the same.